### PR TITLE
build: add Angular 17 support & update packages

### PR DESCRIPTION
**Pull request by Yberion**
Hello @santoshyadavdev,

This PR fixes https://github.com/uiuniversal/ngu-carousel/issues/476.

What I did:

- I couldn't use `yarn` at all this time, I switched to `npm` (even after everything was updated with `npm` and switching back to `yarn`)
- I updated all packages to latest versions (except `jasmine-core`)
- The nx cache is now located in the root folder of the repo, I added `.nx` in `.gitignore`
- There was some problem with how the tsconfigs were done, I fixed all of them (it was causing problems with nx migrations)
- I applied the migrations from nx
- I migrated `ngu-carousel-example` to use the new control-flow
  - I did not update the `README.md`
- I removed the cypress component testing config from `ngu-carousel-example` as it was more or less useless
- I fixed errors found by the linter
- I fixed unit tests
- I removed `updateBuildableProjectDepsInPackageJson` from `ngu-carousel` as it was deprecated
  - removing it keeps the same behavior as if it was enable, that's what we want
- I added missing `tsconfig.json` (cypress) in the `.eslintrc.json` of `ngu-carousel`
- I updated the current SSR setup
  - I first tried to migrate the new `application` builder, but it was not working properly with modules. ~~I will investigate later if it comes from `nx` or the `Angular CLI`~~ (thanks @JeanMeche for his help)
  - Could be related to https://github.com/angular/angular/issues/52998
  - As said in https://riegler.fr/blog/2023-10-13-v17-builders: 
    - `An interesting point to note is that if either 'ssr' or 'prerender' are set to 'true' in the configuration file, the pages served with 'ng serve' will be generated server side.`
    - This was not working, that's one of the reason I kept the same setup with `ssr-dev-server` & co.
    - I already did the migration to `browser-esbuild` when I migrated to `Angular 16`, so it is already using esbuild.
    - Another reason is that `ng serve` using the ssr is not using `server.ts` (which should not affect this repo, but let's play it safe), track https://github.com/angular/angular-cli/issues/26323
- I cleaned `polyfills` usage
- I updated the `CI` setup
  - Angular supports node starting version `18.13.0`, the CI now runs on `['18.13.0', '20']`
  - Updated to `actions/checkout@v4` & `actions/setup-node@v4`
  - Updated to use `npm`
- I bumped `ngu-carousel` for `Angular 17`
- I bumped `ngu-carousel` to `9.0.0` and I updated `README.md`

TODO on your side:

- Release a new major version `9.0.0`
- Close branches related to package update

Tell me if everything looks ok for you.

Have a nice day.

### CODE REVIEW

File: apps/ngu-carousel-example/src/app/banner/banner.module.ts [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
   NguItemComponent
 } from '@ngu/carousel';
+import { BannerRoutingModule } from './banner-routing.module';
+import { BannerComponent } from './banner.component';
 
 @NgModule({
   declarations: [BannerComponent],
   imports: [
-    CommonModule,
+    AsyncPipe,
```
Comment (JeanMeche) [inline apps/ngu-carousel-example/src/app/banner/banner.module.ts:16]: :+1: It's a good pratice !

File: libs/ngu/carousel/src/test.ts [hunk 2]
```diff
@@ -6,8 +6,8 @@ import {
   platformBrowserDynamicTesting
 } from '@angular/platform-browser-dynamic/testing';
 import 'core-js/stable/reflect';
```
Comment (JeanMeche) [inline libs/ngu/carousel/src/test.ts:8]: Are you sure you still need that core-js import ?

File: apps/ngu-carousel-example/project.json [hunk 3]
```diff
@@ -67,54 +65,45 @@
...
-        "browserTarget": "ngu-carousel-example:build"
+        "buildTarget": "ngu-carousel-example:build"
       }
     },
     "test": {
       "executor": "@angular-devkit/build-angular:karma",
       "options": {
         "main": "apps/ngu-carousel-example/src/test.ts",
-        "polyfills": "apps/ngu-carousel-example/src/polyfills.ts",
+        "polyfills": ["zone.js"],
```
Comment (JeanMeche) [inline apps/ngu-carousel-example/project.json:86]: Should be
```
            "polyfills": [
              "zone.js",
              "zone.js/testing"
            ],
```

File: libs/ngu/carousel/src/test.ts [hunk 4]
```diff
@@ -6,8 +6,8 @@ import {
...
-import 'zone.js/dist/zone';
-import 'zone.js/dist/zone-testing';
+import 'zone.js';
+import 'zone.js/testing';
```
Comment (JeanMeche) [inline libs/ngu/carousel/src/test.ts:10]: These should be provided by the polyfill entry. It's likely you don't need them here.

**Review by JeanMeche — COMMENTED** [review]
Some minor comments !

File: apps/ngu-carousel-example/src/app/wrapped/wrapped-carousel/wrapped-carousel.component.html [hunk 5]
```diff
@@ -1,36 +1,36 @@
...
-        </div>
-      </ng-container>
-    </ngu-tile>
+      <ngu-tile *nguCarouselDef="let item; index as i; let ani = animate" [@slider]="ani">
+        @if (i % 2 === 0) {
+          <ng-container
+            *ngTemplateOutlet="card; context: { $implicit: item, index: i }"
+          ></ng-container>
+        }
+        @if (i % 2 !== 0) {
```
Comment (sheikalthaf) [inline apps/ngu-carousel-example/src/app/wrapped/wrapped-carousel/wrapped-carousel.component.html:15]: You can make it as else condition

**Comment by santoshyadavdev** [comment]
There is a build error.

**Comment by Yberion** [comment]
Hello @santoshyadavdev, the build error seemed to be a problem on netlify's side.

The new pipeline looks fine.

**Comment by santoshyadavdev** [comment]
Thank you @Yberion I have updated the node version it was old, I will run the pipeline again to check.

**Pull request merged**